From 8fa2346723fb74e8220ac9f186dabc2f57e4cb43 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@xilinx.com>
Date: Wed, 15 Jun 2016 10:39:55 -0700
Subject: [PATCH] target-arm: Disable EL3 on A53

The Xilinx kernel doesn't support having EL3 enabled and EL2 disabled.
As EL2 is not yet supported on mainline QEMU disable EL3 until we can
support a complete boot flow.

Signed-off-by: Alistair Francis <alistair.francis@xilinx.com>
Upstream-Status: Inappropriate [workaround feature incompatibility]
---
 target-arm/cpu64.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/target-arm/cpu64.c b/target-arm/cpu64.c
index 1635deb..0896259 100644
--- a/target-arm/cpu64.c
+++ b/target-arm/cpu64.c
@@ -165,7 +165,6 @@ static void aarch64_a53_initfn(Object *obj)
     set_feature(&cpu->env, ARM_FEATURE_V8_SHA256);
     set_feature(&cpu->env, ARM_FEATURE_V8_PMULL);
     set_feature(&cpu->env, ARM_FEATURE_CRC);
-    set_feature(&cpu->env, ARM_FEATURE_EL3);
     cpu->kvm_target = QEMU_KVM_ARM_TARGET_CORTEX_A53;
     cpu->midr = 0x410fd034;
     cpu->revidr = 0x00000000;
-- 
2.7.4

